span.go

```diff
@@ -384,6 +398,25 @@ func (s *Span) ended() bool {
 	return s.SpanData == nil
 }
 
+func (s *Span) setExitSpan(name, spanType string) {
```

Comment:
Looking at the spec, I believe the resource is meant to be set to `subtype || type`.

Subtype and Type can be set after starting a span, so I think we need to do that when ending the span. What we can do is keep track of whether SpanContext.SetDestinationService has been called (even with empty strings), and if it is never called for an exit span, then do this implicit call when ending the exit span.